lib/RequestShortener.js

```diff
@@ -5,39 +5,55 @@
 "use strict";
 
 const path = require("path");
+const NORMALIZE_SLASH_DIRECTION_REGEXP = /\\/g;
+const PATH_CHARS_REGEXP = /[-[\]{}()*+?.,\\^$|#\s]/g;
+const SEPERATOR_REGEXP = /[\/\\]$/;
+const FRONT_OR_BACK_BANG_REGEXP = /^!|!$/;
+
+const normalizeBackSlashDirection = (request) => {
```

Comment:
have you looked into memoizing these functions to see if the help they make a difference?